### PR TITLE
feat: Reentrant variant of `schedStateName`

### PR DESCRIPTION
**Pull request by zfields**

### CODE REVIEW

**Review by rayozzie — CHANGES_REQUESTED** [review]
Please don't do that. 

If this is the right change, then change the base method to be reenteant and change all the callers

**Comment by zfields** [comment]
That was my first instinct, but it does look a little ugly compared to the original.

**Comment by zfields** [comment]
What if it were to have a default value instead of a buffer?

This would allow you to perform `JAToI()`, or a more advance lookup function before entering the function.

File: Application/Framework/sched.c [hunk 1]
```diff
@@ -173,8 +183,11 @@ void schedSetCompletionState(int appID, int successstate, int errorstate)
     if (state[appID].completionSuccessState != successstate || state[appID].completionErrorState != errorstate) {
         state[appID].completionSuccessState = successstate;
         state[appID].completionErrorState = errorstate;
+        char success_state_name[20], error_state_name[20];
```
Comment (m-mcgowan) [inline Application/Framework/sched.c:186]: Would be good with a constant for the maximum state name length.

**Review by m-mcgowan — CHANGES_REQUESTED** [review]

**Pull request merged**